src/model.d.ts

```diff
@@ -765,6 +765,12 @@ export interface IncludeOptions extends Filterable<any>, Projectable, Paranoid {
    * Use sub queries. This should only be used if you know for sure the query does not result in a cartesian product.
    */
   subQuery?: boolean;
+
+  /**
+   * Fields to update if row key already exists (on duplicate key update)? These fields apply to the model being `include`d (only supported by MySQL,
+   * MariaDB, SQLite >= 3.24.0 & Postgres >= 9.5).
+   */
+  updateOnDuplicate?: string[];
```

Comment:
This should not be in `Includeable` but a version of `Includeable` only accepted by methods that accept `updateOnDuplicate`

Reply:
@ephys Sounds good - where should we put this? I'm assuming some form of `implement`ing `Includeable`, but not sure if you've already got code doing that somewhere that I can add to.

Reply:
@ephys ping

Reply:
@ephys Appreciate it. Resolving this, the latest commit should have this change.